drift_dev/lib/src/writer/manager/manager_templates.dart

```diff
@@ -345,4 +372,43 @@ class _ManagerCodeTemplates {
         ))
               );""";
   }
+
+  /// Code for a row class which contains references to other tables
+  String? rowClassWithReferences(
```

Comment:
As I've mentioned in the other PR we should investigate whether we want to perhaps refactor this into records to generate less code and allow destructuring which could enable a nicer API. But I'm fine with this approach as well.

Reply:
Let me take a look

Reply:
Done.
Not sure If like it better or not.

Reply:
I agree it's not that nice with the tests using `.then` but it might be cleaner when doing an assignment in an async/await function?